### PR TITLE
fix(DB/Quest): Rework support for 'Someone Else's Hard Work Pays Off'

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/4566

## Tests Performed:

- 
- 

## How to Test the Changes:

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nyeriah** [comment]
I've identified an issue with the second last power core thing, where the scort npc mentions some mobs/attack wave but nothing actually spawns. Otherwise this works, needs sniff to wrap up that last pack though.

**Comment by Buggyswires** [comment]
Bug confirmed. I did the escort with a group and was the only one actually having the quest.
I didn't die, but the npc still just disappeared at the end without giving credit.

**Comment by Nyeriah** [comment]
> Bug confirmed. I did the escort with a group and was the only one actually having the quest.
> I didn't die, but the npc still just disappeared at the end without giving credit.

This doesn’t relate to this PR I presume

**Comment by Nyeriah** [comment]
Please test this and check if it works, it's still better than a fully broken quest I assume

**Comment by Sneften** [comment]
Tested ✅ 
Got quest complete on 2 characters.

Only thing that's not working is when the groups spawn to ambush you, they just spawn and stand still. Won't enter combat unless you aggro them.
Blizz-like they spawn and run to the escort NPC "forcing" combat.

**Comment by Gultask** [comment]
sql file can't be ran twice:
image

**Comment by Nyeriah** [comment]
> sql file can't be ran twice: image

Correct. SQL that alters table structure can't be ran twice and there's nothing we can do about it

**Review by Gultask — APPROVED** [review]

**Comment by Gultask** [comment]
That's unfortunate :c

**Review by Gultask — CHANGES_REQUESTED** [review]
wait, sorry. It's missing deletes

File: data/sql/updates/pending_db_world/rev_1671290232250846500.sql [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+(19671,14,0,"Ok break time is OVER. Let's go!",12,0,100,1,0,0,17261,0,"Cryo-Engineer Sha'heen"),
+(19671,15,0,"Oh really? And what might that be?",12,0,100,1,0,0,17263,0,"Cryo-Engineer Sha'heen"),
+(19671,16,0,"He was right, you know. I'll have to take that tag-line for my own... It's not like he'll have a use for it anymore!",12,0,100,1,0,0,17264,0,"Cryo-Engineer Sha'heen"),
+(19671,17,0,"Thanks and good luck!",12,0,100,1,0,0,17265,0,"Cryo-Engineer Sha'heen"),
+(19666,0,0,"Bravo! Bravo! Good show... I couldn't convince you to work for me, could I? No, I suppose the needless slaughter of my employees might negatively impact your employment application.",14,0,100,0,0,0,17241,0,"Shadow Lord Xiraxis"),
+(19666,1,0,"Your plan was a good one, Sha'heen, and you would have gotten away with it if not for one thing...",12,0,100,1,0,0,17243,0,"Shadow Lord Xiraxis"),
+(19666,2,0,"Never underestimate the other ethereal's greed!",12,0,100,15,0,0,17244,0,"Shadow Lord Xiraxis");
+
+DELETE FROM `waypoints` WHERE `entry` IN (1967100,1967101,1967102);
+INSERT INTO `waypoints` (`entry`,`pointid`,`position_x`,`position_y`,`position_z`,`point_comment`) VALUES
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1671290232250846500.sql:150]:
```suggestion
DELETE FROM `waypoints` WHERE `entry` IN (19666,1967100,1967101,1967102);
INSERT INTO `waypoints` (`entry`,`pointid`,`position_x`,`position_y`,`position_z`,`point_comment`) VALUES
```

**Review by Gultask — CHANGES_REQUESTED** [review]

**Comment by acidmanifesto** [comment]
Suggestion. Mark as merge later. Then merge before squash day

**Review by Gultask — APPROVED** [review]

**Comment by Gultask** [comment]
> Tested ✅ Got quest complete on 2 characters.
> 
> Only thing that's not working is when the groups spawn to ambush you, they just spawn and stand still. Won't enter combat unless you aggro them. Blizz-like they spawn and run to the escort NPC "forcing" combat.

There's a flag set to make the group aggro the player immediately so I don't know why it's not firing. Maybe changing to script owner will fix it but I've had several problems with that as well

You can try that by changing like 70 to:
```sql
(1967105,9,4,0,0,0,100,0,0,0,0,0,0,107,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Summon Group 0"),
```

**Comment by Nyeriah** [comment]
The amount of time I've available to invest into this rages widely, plus it's SAI and I detest SAI. If anyone wants to try addressing the issues reported, feel free to, you can edit the file as you will. Otherwise I will return to it later, much later

**Comment by Gultask** [comment]
Yeah, don't worry about it. I'm planning on testing it later too, seeing if changing the parameter works :)

File: data/sql/updates/pending_db_world/rev_1671290232250846500.sql [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+(1967106,9,1,0,0,0,100,0,0,0,0,0,0,1,8,0,0,0,0,0,12,1,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Say Line 8"),
+(1967106,9,2,0,0,0,100,0,4000,4000,0,0,0,66,0,0,0,0,0,0,8,0,0,0,0,0,0,0,5.04400,"Cryo-Engineer Sha'heen - On Script - Set Orientation"),
+(1967106,9,3,0,0,0,100,0,1000,1000,0,0,0,66,0,0,0,0,0,0,8,0,0,0,0,0,0,0,0.66322,"Cryo-Engineer Sha'heen - On Script - Set Orientation"),
+(1967106,9,4,0,0,0,100,0,0,0,0,0,0,1,9,0,0,0,0,0,12,1,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Say Line 9"),
+
+(1967107,9,0,0,0,0,100,0,0,0,0,0,0,54,7000,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Pause Waypoint"),
+(1967107,9,1,0,0,0,100,0,0,0,0,0,0,66,0,0,0,0,0,0,8,0,0,0,0,0,0,0,0.13962,"Cryo-Engineer Sha'heen - On Script - Set Orientation"),
+(1967107,9,2,0,0,0,100,0,1000,1000,0,0,0,1,10,0,0,0,0,0,12,1,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Say Line 10"),
+(1967107,9,3,0,0,0,100,0,0,0,0,0,0,17,233,0,0,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Set Emote State 233"),
+(1967107,9,4,0,0,0,100,0,1000,1000,0,0,0,107,1,1,0,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Summon Group 1"),
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1671290232250846500.sql:84]:
```suggestion
(1967107,9,4,0,0,0,100,0,1000,1000,0,0,0,107,1,0,1,0,0,0,1,0,0,0,0,0,0,0,0,"Cryo-Engineer Sha'heen - On Script - Summon Group 1"),
```

**Review by Gultask — COMMENTED** [review]
This works

**Comment by Nyeriah** [comment]
Squash days may or may not happen before the end of times, should we merge?

**Comment by Gultask** [comment]
yeah

**Pull request merged**